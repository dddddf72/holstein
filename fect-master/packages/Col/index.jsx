import { computed, ref, toRefs } from 'vue'
import { createNameSpace, useProvider } from '../utils'
const [createComponent] = createNameSpace('Col')
import './col.less'

const READONLY_LAYOUT_KEY = 'layoutKey'

//calculate tag safe distance

const useDistance = (value) => {
  if (typeof value === 'string') Number(value)

  const tempDistance = 100 / 24
  if (value >= 24) return '100%'
  return `${tempDistance * value}%`
}

export default createComponent({
  props: {
    tag: {
      type: String,
      default: 'div',
    },
    span: {
      type: [String, Number],
      default: 24,
    },
    offset: {
      type: [String, Number],
      default: 0,
    },
  },
  setup(props, { attrs, slots }) {
    const { tag, span, offset } = toRefs(props)
    const gutter = ref(null)
    const { ctx } = useProvider(READONLY_LAYOUT_KEY)
    if (ctx) {
      gutter.value = ctx.gutter.value
    }

    const calcStyle = computed(() => {
      const style = {}
      style.width = useDistance(span.value)
      style.marginLeft = useDistance(offset.value)
      style.paddingLeft = `${gutter.value / 2}px`
      style.paddingRight = style.paddingLeft
      return style
    })

    const safeClass = computed(() => {
      let _class = ''
      if (attrs.class) {
        _class = attrs.class ? attrs.class : ''
        return _class
      }
      _class = attrs.className ? attrs.className : ''
      return _class
    })

    return () => (
      <>
        <tag.value
          {...attrs}
          style={calcStyle.value}
          // ${attrs?.class ? attrs.class : ''}
          className={`fect-col  ${safeClass.value}`}
        >
          {slots.default?.()}
        </tag.value>
      </>
    )
  },
})
