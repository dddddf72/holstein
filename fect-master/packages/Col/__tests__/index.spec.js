import { mount } from '@vue/test-utils'
import Col from '../index'

describe('Col', () => {
  it('should be render as Element', () => {
    const col = mount(Col)
    expect(col.html()).toMatchSnapshot()
    expect(() => col.unmount()).not.toThrow()
  })

  it('should be support tag', () => {
    const wrapper = mount(<Col tag="span"></Col>)
    expect(wrapper.find('span')).toBeTruthy()
  })
})
