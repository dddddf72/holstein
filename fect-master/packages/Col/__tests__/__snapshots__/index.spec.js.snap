// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should be render as Element 1`] = `
<div style="width: 100%; margin-left: 0%; padding-left: 0px; padding-right: 0px;" class="fect-col  ">
  <!---->
</div>
`;
